MigrationPreparer/MigrationPreparerImplementation.cpp

```diff
@@ -0,0 +1,595 @@
+        {
+            LOGINFO("Register notification");
+            _migrationPreparerNotification.push_back(notification);
+            notification->AddRef();
+        }
+
+        _adminLock.Unlock();
+
+        return Core::ERROR_NONE;
+    }
+
+    uint32_t MigrationPreparerImplementation::Unregister(Exchange::IMigrationPreparer::INotification *notification )
+    {
+        uint32_t status = Core::ERROR_GENERAL;
+
+        ASSERT (nullptr != notification);
+
+        _adminLock.Lock();
+
+        // Make sure we can't unregister the same notification callback multiple times
+        auto itr = std::find(_migrationPreparerNotification.begin(), _migrationPreparerNotification.end(), notification);
+        if (itr != _migrationPreparerNotification.end())
+        {
+            (*itr)->Release();
+            LOGINFO("Unregister notification");
+            _migrationPreparerNotification.erase(itr);
+            status = Core::ERROR_NONE;
+        }
+        else
+        {
+            LOGERR("notification not found");
+        }
+
+        _adminLock.Unlock();
+
+        return status;
+    }
+
+    void MigrationPreparerImplementation::ValueChanged(const string &name, const string &value)
+    {
+         LOGINFO("name:%s value:%s",name.c_str(), value.c_str());
+         _adminLock.Lock();
+         std::list<Exchange::IMigrationPreparer::INotification*>::const_iterator index(_migrationPreparerNotification.begin());
+         while (index != _migrationPreparerNotification.end())
+         {
+            (*index)->ValueChanged(name.c_str(), value.c_str());
+            index++;
+         }
+         _adminLock.Unlock();         
+    }
+
+    /*Helper's: Begin*/
+    string MigrationPreparerImplementation::escapeSed(string input, enum sedType type) {
+        // Set of special characters in sed that need to be escaped
+        std::unordered_set<char> specialChars = {
+            '.', '*', '+', '?', '^', '$', '(', ')', '[', ']', '{', '}', '\\', '|', '/', '&'
+        };
+
+        std::string escapedString;
+
+        if(type == PATTERN) {
+            // Iterate through each character in the string
+            for(char ch : input) {
+                // If ch is a special character replace with .
+                if (specialChars.find(ch) != specialChars.end())
+                    escapedString += ".";
+                else
+                    escapedString += ch;
+            }
+            escapedString = "\":"+ escapedString;
+        }
+        else if(type == REPLACEMENT) {
+            // Iterate through each character in the string
+            for(char ch : input) {
+                // If ch is a special character prepend it with a backslash
+                if (specialChars.find(ch) != specialChars.end())
+                    escapedString += "\\";
+
+                escapedString += ch;
+            }
+            escapedString = "\":"+ escapedString;
```

Comment:
Since while creating a PATTERN string for substitution as stated above, we append ": to the beginning of the string. While creating REPLACEMENT string we have add ": to the beginning.  Otherwise ": will be missing.
say we have 
{
"key":"val"
}
if we don't add :" to the REPLACEMENT string it would like below
{
"key"updated-val"
}